Scripts/Modules/AdaptivePositionModule/AdaptivePositionModule.cs

```diff
@@ -0,0 +1,131 @@
+﻿#if UNITY_EDITOR && UNITY_2017_2_OR_NEWER
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.Experimental.EditorVR.Core;
+using UnityEditor.Experimental.EditorVR.Extensions;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR.Modules
+{
+    public sealed class AdaptivePositionModule : MonoBehaviour, IDetectGazeDivergence, IUsesViewerScale, IControlHaptics
+    {
+        [SerializeField]
+        HapticPulse m_MovingPulse; // The pulse performed while moving an element to a new target position in the user's FOV
+
+        Coroutine m_AdaptiveElementRepositionCoroutine;
+        bool m_TestInFocus;
+        Transform m_GazeTransform;
+        Transform m_WorldspaceAnchorTransform; // The player transform under which anchored objects will be parented
+
+        // Collection objects whose position is controlled by this module
+        readonly List<IAdaptPosition> m_AdaptivePositionElements = new List<IAdaptPosition>();
+
+        public class AdaptivePositionData
+        {
+            public AdaptivePositionData(IAdaptPosition caller, Vector3 previousAnchoredPosition)
+            {
+                this.previousAnchoredPosition = previousAnchoredPosition;
+                startingPosition = caller.adaptiveTransform.position;
+            }
+
+            /// <summary>
+            /// The world-position at which this object was last anchored
+            /// </summary>
+            public Vector3 previousAnchoredPosition { get; set; }
+
+            /// <summary>
+            /// The origin/starting position of the object being re-positioned
+            /// </summary>
+            public Vector3 startingPosition { get; set; }
+
+        }
+
+        void Awake()
+        {
+            m_GazeTransform = CameraUtils.GetMainCamera().transform;
+            m_WorldspaceAnchorTransform = m_GazeTransform.parent;
+        }
+
+        void Update()
+        {
+            if (m_AdaptivePositionElements.Count > 0)
+            {
+                var adaptiveElement = m_AdaptivePositionElements.First();
```

Comment:
Why just .first()? Also, this will allocate. Not a huge deal but we can do a [0] instead since it's a list.